pkgs/games/ldmud/default.nix

```
@@ -0,0 +1,79 @@
+{ lib, fetchFromGitHub, stdenv, autoreconfHook, pkg-config, bison, libiconv
+, pcre, libgcrypt, ipv6Support ? false, mccpSupport ? false, zlib
+, mysqlSupport ? false, libmysqlclient, postgresSupport ? false, postgresql
+, sqliteSupport ? false, sqlite, json_c, libxml2, tlsSupport ? false, openssl
+, pythonSupport ? false, python310 }:
+
+stdenv.mkDerivation rec {
+  pname = "ldmud";
+  version = "3.6.6";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-2TaFt+T9B5Df6KWRQcbhY1E1D6NISb0oqLgyX47f5lI=";
+  };
+
+  sourceRoot = "${src.name}/src";
+
+  nativeBuildInputs =
+    [ autoreconfHook pkg-config bison libgcrypt pcre json_c libxml2 ]
+    ++ lib.optional mccpSupport zlib ++ lib.optional mysqlSupport libmysqlclient
+    ++ lib.optional postgresSupport postgresql
+    ++ lib.optional sqliteSupport sqlite ++ lib.optional tlsSupport openssl
+    ++ lib.optional pythonSupport python310;
+  buildInputs = lib.optionals stdenv.isDarwin [ libiconv ];
+
+  # To support systems without autoconf LD puts its configure.ac in a non-default
+  # location and uses a helper script. We skip that script and symlink the .ac
+  # file to where the autoreconfHook find it.
+  preAutoreconf = ''
+    ln -fs ./autoconf/configure.ac ./
+  '';
+
+  configureFlags = [
+    "--enable-erq=xerq"
+    "--enable-filename-spaces"
+    "--enable-use-json"
+    "--enable-use-xml=xml2"
+    (lib.enableFeature ipv6Support "use-ipv6")
+    (lib.enableFeature mccpSupport "use-mccp")
+    (lib.enableFeature mysqlSupport "use-mysql")
+    (lib.enableFeature postgresSupport "use-pgsql")
+    (lib.enableFeature sqliteSupport "use-sqlite")
+    (lib.enableFeatureAs tlsSupport "use-tls" "ssl")
+    (lib.enableFeature pythonSupport "use-python")
+  ];
+
+  preConfigure = lib.optionalString mysqlSupport ''
+    export CPPFLAGS="-I${lib.getDev libmysqlclient}/include/mysql"
+    export LDFLAGS="-L${libmysqlclient}/lib/mysql"
+  '';
+
+  installTargets = [ "install-driver" "install-utils" "install-headers" ];
+
+  postInstall = ''
+    mkdir -p "$out/share/"; cp -v ../COPYRIGHT $out/share/;
+  '';
+
+  meta = with lib; {
+    description =
+      "LDMud is a gamedriver for LPMuds. (LPC compiler, interpreter and runtime environment.)";
```

See `CONTRIBUTING.md`.
```suggestion
    description = "A gamedriver for LPMuds including a LPC compiler, interpreter and runtime";
```